rest-service/manager_rest/storage/management_models.py

```diff
@@ -37,28 +37,47 @@ def _get_response_data(resources, get_data=False, name_attr='name'):
     """Either return the sorted list of resource names or their total count
 
     :param resources: A list/dict of users/tenants/user-groups
-    :param name_attr: The name attribute (name/username)
+    :param name_attr:
+        The name attribute (name/username) or a dictionary where 'key' is the
+        attribute name for the keys and 'value' is the attribute name for the
+        values.
     :param get_data: If True: return the names, o/w return the count
     """
     if get_data:
         if isinstance(resources, list):
             return sorted(getattr(res, name_attr) for res in resources)
         elif isinstance(resources, dict):
+            if isinstance(name_attr, str):
```

Comment:
I'm really starting not to like how this is done. It is _really_ difficult to follow this function. I wish there was a cleaner way to handle getting response data.

Reply:
Yes, I agree. I'm going to merge for now, but I'll look at it again later to refactor it.